src/image_handler.cpp

```diff
@@ -117,6 +122,138 @@ static uint16_t mapColorValue(uint8_t pixelColor, uint16_t color2, uint16_t colo
   }
 }
 
+///////////////////////////////////////////////
+// Direct Streaming Context
+///////////////////////////////////////////////
+
+#if defined(STREAMING_ENABLED) && defined(STREAMING_DIRECT_MODE)
+
+// Context for direct streaming decode
+struct DirectStreamContext
+{
+  StreamingHandler::RowStreamBuffer *buffer;
+  uint16_t displayWidth;
+  uint16_t displayHeight;
+  uint16_t currentRow;       // Current absolute row being decoded
+  uint16_t bufferRowIndex;   // Index within the row buffer (0 to bufferRowCount-1)
+  uint16_t bufferRowCount;   // Number of rows in buffer
+  uint16_t firstRowInBuffer; // First absolute row number in current buffer
+  uint32_t pixelsProcessed;
+  bool initialized;
+};
+
+static DirectStreamContext g_directCtx = {nullptr, 0, 0, 0, 0, 0, 0, 0, false};
+
+// Flush completed rows from buffer to display
+static void flushCompletedRows()
+{
+  if (!g_directCtx.initialized || !g_directCtx.buffer)
+    return;
+
+  // bufferRowIndex is the index of the last row written (0-based)
+  // We need to flush bufferRowIndex + 1 rows (indices 0 through bufferRowIndex)
+  uint16_t rowsToFlush = g_directCtx.bufferRowIndex + 1;
+  if (g_directCtx.bufferRowIndex == 0 && g_directCtx.pixelsProcessed == 0)
+    return; // No rows to flush
```

Comment:
I would say If `bufferRowIndex` is 0 but pixels WERE processed, it won't flush. Maybe this will fix it?

```suggestion
  // Check if current row has any pixels written
  uint16_t currentRowPixels = g_directCtx.buffer->getRowPixelCount(g_directCtx.bufferRowIndex);
  if (currentRowPixels == 0 && g_directCtx.bufferRowIndex == 0)
    return; // Nothing to flush

  uint16_t rowsToFlush = g_directCtx.bufferRowIndex + 1;
```